src/action/common/provision_nix.rs

```diff
@@ -157,19 +167,51 @@ impl Action for ProvisionNix {
 /// If there is an existing /nix/store directory, ensure that the group ID we're going to use for
 /// the nix build group matches the group that owns /nix/store to prevent weird mismatched-ownership
 /// issues.
-async fn check_existing_nix_store_gid_matches(
-    desired_nix_build_group_id: u32,
-) -> Result<(), ActionErrorKind> {
+async fn ensure_nix_store_group(desired_nix_build_group_id: u32) -> Result<(), ActionErrorKind> {
     let previous_store_metadata = tokio::fs::metadata(NIX_STORE_LOCATION)
         .await
         .map_err(|e| ActionErrorKind::GettingMetadata(NIX_STORE_LOCATION.into(), e))?;
     let previous_store_group_id = previous_store_metadata.gid();
     if previous_store_group_id != desired_nix_build_group_id {
-        return Err(ActionErrorKind::PathGroupMismatch(
-            NIX_STORE_LOCATION.into(),
-            previous_store_group_id,
-            desired_nix_build_group_id,
-        ));
+        for (entry, _metadata) in walkdir::WalkDir::new(NIX_STORE_LOCATION)
+        .follow_links(false)
+        .same_file_system(true)
+        // chown all of the contents of the dir before NIX_STORE_LOCATION,
+        // this means our test of "does /nix/store have the right gid?"
+        // is useful until the entire store is examined
+        .contents_first(true)
+        .into_iter()
```

Comment:
this formatting is kinda funky... I think it's probably because of the long `tracing::warn`s -- could you try breaking those and see if that makes it better?